### PR TITLE
[SPARK-35013][CORE] Don't allow to set spark.driver.cores=0

### PR DESCRIPTION
**Pull request by shahidki31**
### What changes were proposed in this pull request?
Currently spark is not allowing to set spark.driver.memory, spark.executor.cores, spark.executor.memory to 0, but allowing driver cores to 0. This PR checks for driver core size as well. Thanks @Oleg Lypkan for finding this.

### Why are the changes needed?
To make the configuration check consistent.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manual testing

### CODE REVIEW

**Comment by shahidki31** [comment]
Sure @dongjoon-hyun . Let me go through the doc. Thanks

**Comment by shahidki31** [comment]
@dongjoon-hyun I have updated the branch with master. Kindly review

**Comment by HyukjinKwon** [comment]
The tests seem failed. Would you please fixing them?

**Comment by HyukjinKwon** [comment]
@shahidki31 can you retrigger the test please? https://github.com/shahidki31/spark/actions/runs/832876763

**Comment by HyukjinKwon** [comment]
cc @Ngone51 @mridulm @tgravescs FYI

**Comment by shahidki31** [comment]
> https://github.com/shahidki31/spark/actions/runs/832876763

Sure @HyukjinKwon . Let me check

**Comment by shahidki31** [comment]
> @shahidki31 can you retrigger the test please? https://github.com/shahidki31/spark/actions/runs/832876763
 Sure @HyukjinKwon Let me check

**Comment by shahidki31** [comment]
Hi @HyukjinKwon All the tests are passing. Seems Linters, Licence job is failing. But that is unrelated to the PR. Is it a known failure?
Screenshot 2021-05-12 at 12 30 05 PM

**Comment by HyukjinKwon** [comment]
That's fixed in the upstream. Can you sync with the latest master branch of Apache Spark repo by rebasing/merging?

**Comment by shahidki31** [comment]
Thanks. Let me retrgger that.

**Review by Ngone51 — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala [hunk 1]
```diff
@@ -253,6 +253,9 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
         && Try(JavaUtils.byteStringAsBytes(executorMemory)).getOrElse(-1L) <= 0) {
       error("Executor memory must be a positive number")
     }
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala:255]:
```suggestion
    }
```

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala [hunk 2]
```diff
@@ -253,6 +253,9 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
...
+    if (driverCores != null && Try(driverCores.toInt).getOrElse(-1) <= 0) {
+      error("Driver cores must be a positive number")
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala:257]:
```suggestion
       error("Driver cores must be a positive number")
```

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala [hunk 3]
```diff
@@ -253,6 +253,9 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
         && Try(JavaUtils.byteStringAsBytes(executorMemory)).getOrElse(-1L) <= 0) {
       error("Executor memory must be a positive number")
     }
+    if (driverCores != null && Try(driverCores.toInt).getOrElse(-1) <= 0) {
+       error("Driver cores must be a positive number")
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala:257]:
```suggestion
      error("Driver cores must be a positive number")
```

**Comment by tgravescs** [comment]
changes look fine, assume it was manually tested

**Comment by shahidki31** [comment]
> changes look fine, assume it was manually tested

Yes, I verified manually
image

**Comment by shahidki31** [comment]
Retest this please

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Thank you, @shahidki31 and all.
Merged to master.

**Comment by shahidki31** [comment]
Thanks @dongjoon-hyun and all for the reviews

**Pull request closed without merging**